src/elm/Ui/Input.elm

```diff
@@ -0,0 +1,64 @@
+module Ui.Input exposing
+    ( radio
+    , radioGroup
+    )
+
+import Html as H exposing (Attribute, Html)
+import Html.Attributes as A
+import Html.Events as E
+import Html.Extra exposing (viewMaybe)
+import Ui.Button exposing (ButtonMode(..))
+import Ui.Group
+import Ui.TextContainer as Text exposing (TextColor(..), TextContainer(..))
+
+
+type alias RadioOptions msg =
```

Comment:
This is quite big, have you thought about making this component into a builder pattern? So an `init` function that returns a default value, and then setters to change the fields, and then finally a `view` function. In use, you would end up with something like:

```elm
Radio.init
    |> Radio.setTitle "My radio"
    |> Radio.setChecked True
    |> Radio.setOnCheck myOnCheck
    |> Radio.view
```

The `init` function can of course take things that are _always_ needed or useful as input, e.g. in this case it might not make sense to not have a title.

Another huge benefit with the builder pattern is that you can expand this record with more fields without having to update the entire codebase.

Reply:
I was the builder pattern was used in Notification, but noticed it too late. I decided on using record types early on to prevent ambiguous order and easier expanding (also per Rich Hickey's teachings 😬 ). I can explore rewriting this

Reply:
Oh yeah, don't get me wrong. Using records over tuples or a bazillion function arguments is really good, it makes it much easier to deal with. Rewriting it isn't strictly needed in this PR, but something to think about.

Reply:
https://github.com/AtB-AS/webshop/issues/42